### ISSUE TITLE
Migrating from Local to WcfProxy only a subset of my business objects

### ISSUE THREAD

**Issue by GillesBer** [MarimerLLC/cslaforum#475]
Dear all,
I'm running and maintaining a large 2-tier application that is using CSLA 4.6.600.
We would like to migrate this application to 3-tier, but in several phases.
Is there any 'best practice' to migrate only a subset of my business objects?

Our plan was to migrate: 'one form at a time', and their business objects.

I imagine that I can also add the [RunLocal] attribute on all my legacy objets to 'opt-out' them from 3 tier. But ideally, I would like to 'opt-in' my new forms (and related business objects) and migrate my business objects 'form' by 'form'.

I found this old 2008 thread on http://forums.lhotka.net/forums/p/4413/21609.aspx#21609 which proposed to choose to which Proxy it redirects. I imagine this is still valid solution.
Has someone any experience doing so? Are there any pitfall I should anticipate.
Regards, 

Gilles

**Comment by rockfordlhotka** [comment]
Yes, you'll need to create a custom client-side data portal proxy that uses an attribute (or something) to determine which objects should run local vs remote.
